Controllers/FileController.cs

```diff
@@ -23,27 +80,91 @@ public async Task<ActionResult> Post()
                 Directory.CreateDirectory(uploadPath);
             }
 
-            var path = Path.Combine(uploadPath, file.FileName);
+            foreach (var file in files)
+            {
+                var path = Path.Combine(uploadPath, file.FileName);
+
+                using (var stream = new FileStream(path, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+            }
+            string uploaderName = form["uploader"];
+            string desserts = form["dessert"];  // Assuming this is a string like "dessert1,dessert2,dessert3"
+            string weather = form["weather"];
+            string country = form["country"];
+
+            // Split the desserts string into an array of individual desserts
+            string[] dessertArray = desserts.Split(',');
```

Comment:
returns a 500 when there isn't a dessert vote present, should either return a 400, or accept without a dessert vote